specs/eid-hermes-commands-format.md

```diff
@@ -0,0 +1,227 @@
+
+**Table 3: Status**
+
+## List of Commands
+
+The following commands are defined for Hermes devices:
+
+| Opcode | Command        |
+|--------|----------------|
+| 0x00   | Request Slot   |
+| 0x01   | Release Slot   |
+| 0x10   | Write to Slot  |
+| 0x11   | Read from Slot |
+| 0x80   | Run Program    |
+
+**Table 4: List of commands**
+
+### Request Slots (Opcode 0x00)
+
+This command can be used to reserve a program or data slot on the device.
+
+| Bytes | Description                           |
+|-------|---------------------------------------|
+| 08    | Slot type (0x00: program, 0x01: data) |
+
+**Table 5: Request Slots Command Request**
+
+| Bytes | Description                                          |
+|-------|------------------------------------------------------|
+| 08    | ID of allocated slot. Only valid when Status is 0x00 |
+
+**Table 6: Request Slots Command Response**
+
+This command may return the following status:
+
+| Status | Description                            |
+|--------|----------------------------------------|
+| 0x00   | Success                                |
+| 0x01   | All slots of requested type are in use |
+
+**Table 7: Request Slots status codes**
+
+### Release Slots (Opcode 0x01)
+
+This command can be used to release a program or data slot on the device.
+
+It is invalid to release a slot that is not allocated.
+
+| Bytes | Description                           |
+|-------|---------------------------------------|
+| 08    | Slot type (0x00: program, 0x01: data) |
+| 09    | Slot ID                               |
+
+**Table 8: Release Slots Command Request**
+
+| Bytes | Description                |
+|-------|----------------------------|
+
+(no command-specific bytes used)
+
+**Table 9: Release Slots Command Response**
+
+This command may return the following status:
+
+| Status | Description                                                |
+|--------|------------------------------------------------------------|
+| 0x00   | Success                                                    |
+| 0x02   | Requested program slot does not exist or is not allocated  |
+| 0x03   | Requested data slot does not exist or is not allocated     |
+
+**Table 10: Release Slots status codes**
+
+### Write to Slot (Opcode 0x10)
+
+This command can be used to write to a program or data slot. The slot must have
+been previously requested with opcode 0x00 or this will fail.
+
+This command does not guarantee that all bytes will be transferred. The number
+of actual bytes transferred is reported on the Command Response.
+
+| Bytes | Description                               |
+|-------|-------------------------------------------|
+| 08    | Slot type (0x00: program, 0x01: data)     |
+| 09    | Slot ID                                   |
+| 11:10 | Reserved                                  |
+| 19:12 | Source Addr: the host address to transfer |
+| 23:20 | Data Length                               |
```

Comment:
Is data length in Bytes? If so we should make that clear.

Reply:
Yes, bytes. Will fix